ACE/ace/os_include/os_pthread.h

```diff
@@ -45,9 +45,13 @@
 
 
 #if !defined (ACE_LACKS_PTHREAD_H)
+# if !defined ACE_VXWORKS || ACE_VXWORKS < 0x700
```

Comment:
The original code was always adding "extern C" which makes a bad assumption.  It would be nice to have a configuration macro so that only platforms that need an "extern C" would get one.  That way when there are 0 such platforms the code should be removed.  VxWorks7 seems to be the only platform that actually breaks with it there.